### PR TITLE
Fix compilation warnings  #494

### PR DESCRIPTION
**Pull request by rpkarlsson**
Fix compilation warnings as stated in #494.

-----------------
- [X] The commits are consistent with our [contribution guidelines][1].
- [ ] You've added tests (if possible) to cover your change(s). Bugfix, indentation, and font-lock tests are extremely important!
- [X] You've run `M-x checkdoc` and fixed any warnings in the code you've written.
- [ ] You've updated the changelog (if adding/changing user-visible functionality).
- [ ] You've updated the readme (if adding/changing user-visible functionality).

[1]: https://github.com/clojure-emacs/clojure-mode/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: clojure-mode.el [hunk 1]
```diff
@@ -2063,9 +2056,8 @@ many times."
         (condition-case nil
             (save-match-data
               (let ((original-position (point))
-                    clojure-comment-start clojure-comment-end)
+                    clojure-comment-end)
                 (beginning-of-defun)
-                (setq clojure-comment-start (point))
```
Comment (bbatsov) [inline clojure-mode.el]: Why did you change this?

Comment (rpkarlsson) [reply clojure-mode.el]: `clojure-comment-start` was generating a warning and as far as I can tell It is unused? It may be that I'm missunderstanding something?
```
clojure-mode.el:2023:1:Warning: Unused lexical variable
    ‘clojure-comment-start’
```

Comment (bbatsov) [reply clojure-mode.el]: Hmm, I think that's supposed to be a buffer local variable. Let me take a closer look at the code - we might have made some real mistake. :-)

**Comment by bbatsov** [comment]
Sorry about the long radio silence! After a closer inspection of the changes I feel they are all safe. Thanks!
